src/test/java/com/hyperwallet/clientsdk/util/HyperwalletApiClientTest.java

```diff
@@ -1273,234 +1260,199 @@ public void testPost_200Response_ResponseContentTypeHeaderWithLeadingCharset() {
+        hyperwalletDocument.setUploadFiles(uploadFiles);
+        List<HyperwalletVerificationDocument> hyperwalletDocumentList = new ArrayList<>();
+        hyperwalletDocumentList.add(hyperwalletDocument);
+        hyperwalletUser.setDocuments(hyperwalletDocumentList);
+        HashMap<String,String> multiPartUploadData = new HashMap<String,String>();
+
+        HyperwalletApiClient mockApiClient = createAndInjectHyperwalletApiClientMock(client);
+        Mockito.when(mockApiClient.put(Mockito.anyString(), Mockito.any(HyperwalletUser.class), Mockito.any(Class.class)))
+                .thenReturn(hyperwalletUser);
+        HyperwalletUser hyperwalletUserresponse = client.uploadUserDocuments("test-token", hyperwalletDocumentList);
+        assertThat(hyperwalletUserresponse, isOneOf(null, hyperwalletUser));
+    }
 
-            FormDataMultiPart multiPart = new FormDataMultiPart();
+    @Test
+    public void testDocumentUploadWithError() {
+        try {
             JSONObject jsonObject = new JSONObject();
-            jsonObject.put("type", "LETTER_OF_AUTHORIZATION");
-            jsonObject.put("category", "AUTHORIZATION");
+            jsonObject.put("type", "DRIVERS_LICENSE");
+            jsonObject.put("country", "AL");
+            jsonObject.put("category", "IDENTIFICATION");
             List<JSONObject> jsonObjectList = new ArrayList<>();
             jsonObjectList.add(jsonObject);
-
             JSONObject jsonObject1 = new JSONObject();
             jsonObject1.put("documents", jsonObjectList);
-            BodyPart data =
-                    new FormDataBodyPart(FormDataContentDisposition.name("data").build(), jsonObject1.toString(), MediaType.APPLICATION_JSON_TYPE);
-            multiPart.bodyPart(data);
-
-            Client mockClient = createAndInjectWebResourceClient(hyperwalletApiClientEnc);
-
-            WebResource webResource = mock(WebResource.class);
-            when(mockClient.resource(baseUrl + "/documentUpload")).thenReturn(webResource);
-            WebResource.Builder builder = mock(WebResource.Builder.class);
-            when(webResource.type(MediaType.MULTIPART_FORM_DATA_TYPE)).thenReturn(builder);
-
-            InBoundHeaders headers = new InBoundHeaders();
-            headers.put(HttpHeaders.CONTENT_TYPE, Collections.singletonList("application/jose+json"));
-
-            ClientResponse clientResponse = mock(ClientResponse.class);
-
-            when(builder.put(ClientResponse.class, multiPart)).thenReturn(clientResponse);
-            when(clientResponse.getStatus()).thenReturn(200);
-
-            when(clientResponse.getEntity(String.class)).thenReturn("{result: \"Success\"}");
-            when(clientResponse.getHeaders()).thenReturn(headers);
-
-            hyperwalletApiClientEnc.put(baseUrl + "/documentUpload", multiPart, HyperwalletUser.class);
+            HyperwalletVerificationDocument multiPartUploadData = new HyperwalletVerificationDocument();
```

Comment:
No specific business logic in this class. ApiClient has no idea about HyperwalletVerificationDocument. Only Multipart object.